sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/implementation/PersonalizerClientV1Preview3ImplBuilder.java

```diff
@@ -230,70 +263,63 @@ public PersonalizerClientV1Preview3ImplBuilder retryPolicy(RetryPolicy retryPoli
 
     /**
      * Builds an instance of PersonalizerClientV1Preview3Impl with the provided parameters.
-     *
+     * 
      * @return an instance of PersonalizerClientV1Preview3Impl.
      */
     @Generated
     public PersonalizerClientV1Preview3Impl buildClient() {
-        if (pipeline == null) {
-            this.pipeline = createHttpPipeline();
-        }
-        if (apiVersion == null) {
-            this.apiVersion = "v1.1-preview.3";
-        }
-        if (serializerAdapter == null) {
-            this.serializerAdapter = JacksonAdapter.createDefaultSerializerAdapter();
-        }
-        PersonalizerClientV1Preview3Impl client =
-                new PersonalizerClientV1Preview3Impl(pipeline, serializerAdapter, endpoint, apiVersion);
+        this.validateClient();
+        HttpPipeline localPipeline = (pipeline != null) ? pipeline : createHttpPipeline();
+        SerializerAdapter localSerializerAdapter
+            = (serializerAdapter != null) ? serializerAdapter : JacksonAdapter.createDefaultSerializerAdapter();
+        PersonalizerClientV1Preview3Impl client = new PersonalizerClientV1Preview3Impl(localPipeline,
+            localSerializerAdapter, this.endpoint, this.apiVersion);
         return client;
     }
 
+    @Generated
+    private void validateClient() {
+        // This method is invoked from 'buildInnerClient'/'buildClient' method.
+        // Developer can customize this method, to validate that the necessary conditions are met for the new client.
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        Objects.requireNonNull(apiVersion, "'apiVersion' cannot be null.");
```

Comment:
I am not sure why this is a required field. There should be a default if it's not set.

Reply:
I've seen this elsewhere in code generation, might be a case where the REST API version is part of the path doesn't get handled as well.